### PR TITLE
5757 NR Results pdf download

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#5757

*Description of changes:*

* Download PDF outputs for a given NR iD

* Loading Indicator

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: client/src/components/existing-request/existing-request-display.vue [hunk 1]
```diff
@@ -528,6 +530,9 @@ export default class ExistingRequestDisplay extends Mixins(
         case NrAction.RETRY_PAYMENT:
           this.navigateToPaymentPortal()
           break
+        case NrAction.RESULTS:
```
Comment (severinbeauvais) [inline client/src/components/existing-request/existing-request-display.vue:533]: Does this button have a loading indicator while the results are being created/downloaded?

File: client/src/components/mixins/output-mixin.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      const headers = { 'Accept': 'application/pdf' }
+
+      // show spinner since the network calls below can take a few seconds
+      this.$root.$emit('showSpinner', true)
+
+      // Request PDF for specified id
+      const response: any = await axios.get(url, { headers: headers, responseType: 'blob' as 'json' })
+
+      // Create a new blob object with mime-type explicitly set, otherwise only Chrome works
+      const blob = new Blob([response.data], { type: 'application/pdf' })
```
Comment (severinbeauvais) [inline client/src/components/mixins/output-mixin.ts:22]: I think this code already exists in one of the services/modules. Can you just reuse it?

**Review by lekshmimallika-aot — APPROVED** [review]

File: client/src/components/existing-request/existing-request-display.vue (hunk 1 above)
Comment (cameron-freshworks) [reply client/src/components/existing-request/existing-request-display.vue:533]: I utilized the global spinner opposed to locking out the individual button. I figured since there was multiple actions a user could click on, the global spinner was a good choice. Any thoughts on this?

File: client/src/components/mixins/output-mixin.ts (hunk 2 above)
Comment (cameron-freshworks) [reply client/src/components/mixins/output-mixin.ts:22]: ah interesting! I'll see if i can find this.

Comment (cameron-freshworks) [reply client/src/components/mixins/output-mixin.ts:22]: I see some code that is related to payments for receipts, it is different however. Is this the code you were referring too?

File: client/src/components/existing-request/existing-request-display.vue (hunk 1 above)
Comment (severinbeauvais) [reply client/src/components/existing-request/existing-request-display.vue:533]: Ah, that works.

I have no preference, other than having a common mechanism -- but which is the common mechanism in NR?

TLDR - you don't need to change this (unless, say, Scott/Tracey have a preference).

Comment (severinbeauvais) [reply client/src/components/existing-request/existing-request-display.vue:533]: PS I think the UI would support downloading the results AND doing something else concurrently (unless there are global state conflicts). Again, I have no pref whether to block just this button or the whole UI.

PPS What does the error handling look like?

File: client/src/components/mixins/output-mixin.ts (hunk 2 above)
Comment (severinbeauvais) [reply client/src/components/mixins/output-mixin.ts:22]: That sounds like it. Doesn't it just download a PDF? I would rather see a single method for download a PDF. Maybe the receipt code needs to be refactored to use a common download method?

File: client/src/components/existing-request/existing-request-display.vue (hunk 1 above)
Comment (cameron-freshworks) [reply client/src/components/existing-request/existing-request-display.vue:533]: Oh, interesting, right!
Currently were logging the error but i will add something in for the UI.
ie a modal, although i didn't see a design for that. Possibly use a generic confirm, as a failure would/could just be tried again with no detriment.

File: client/src/components/mixins/output-mixin.ts (hunk 2 above)
Comment (cameron-freshworks) [reply client/src/components/mixins/output-mixin.ts:22]: I think if it was just hitting a different endpoint, making it generic could work but the request for outputs was purposely removed from the `/payments` endpoint and payment module. 
Imo, i think we leave payments ( receipts ) standalone, but in the future if we find we need a generic download, we modify this mixin method to handle that, is that ok?

File: client/src/components/existing-request/existing-request-display.vue (hunk 1 above)
Comment (severinbeauvais) [reply client/src/components/existing-request/existing-request-display.vue:533]: A simple error modal would be OK. The user can simply retry from the main page.

Comment (severinbeauvais) [reply client/src/components/existing-request/existing-request-display.vue:533]: (The error modal may already display.)

Comment (cameron-freshworks) [reply client/src/components/existing-request/existing-request-display.vue:533]: Added! Thanks, good catch 😁

File: client/src/components/mixins/output-mixin.ts (hunk 2 above)
Comment (severinbeauvais) [reply client/src/components/mixins/output-mixin.ts:24]: OK, but could you move the network code to one of the 2 places where there is other network code?

Or do you feel strongly that we should update the architecture to have separate mixins for separate actions?

Comment (cameron-freshworks) [reply client/src/components/mixins/output-mixin.ts:24]: I don't feel strongly enough about it :) We can move it anywhere, although i prefer the mixin approach. 
What are the options, i don't see a place where it really fits other than the `newRequestModule`?
The requests made from `existing-Request-Display` refer to `newrequestModule`, so i guess that's the most logical.
(Although NOT a new request at this point 😆 )

Comment (cameron-freshworks) [reply client/src/components/mixins/output-mixin.ts]: kk, relocated!

**Review by severinbeauvais — APPROVED** [review]
Looks good.
